test/functional/kernel/tile-variants/tests/test-kernel-tile-Fixed2DSum.hpp

```diff
@@ -0,0 +1,143 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_KERNEL_TILE_FIXED2DSUM_HPP__
+#define __TEST_KERNEL_TILE_FIXED2DSUM_HPP__
+
+#include <numeric>
+#include <vector>
+#include <type_traits>
+
+// Remove testing of float for OpenMP and CUDA.
+// OpenMP collapse and CUDA policies produces inaccurate results for float type.
+// Instruction ordering seems to matter for these policies on floats.
+template <typename INDEX_TYPE, typename DATA_TYPE, typename WORKING_RES, typename EXEC_POLICY, typename REDUCE_POLICY>
+typename std::enable_if<
+           (std::is_same<DATA_TYPE,float>::value &&
+            (
+              #if defined(SYCL_LANGUAGE_VERSION) && defined(__INTEL_LLVM_COMPILER)
+              true  // represents SYCL compiler
+              #elif defined(__INTEL_COMPILER)
+              true  // represents Intel compiler
+              #elif defined(RAJA_ENABLE_OPENMP) && defined(RAJA_ENABLE_CUDA)
+              std::is_same<REDUCE_POLICY,RAJA::omp_reduce>::value || std::is_same<REDUCE_POLICY,RAJA::cuda_reduce>::value
+              #elif defined(RAJA_ENABLE_OPENMP) && defined(RAJA_ENABLE_HIP)
+              std::is_same<REDUCE_POLICY,RAJA::omp_reduce>::value || std::is_same<REDUCE_POLICY,RAJA::hip_reduce>::value
+              #elif defined(RAJA_ENABLE_OPENMP)
+              std::is_same<REDUCE_POLICY,RAJA::omp_reduce>::value
+              #elif defined(RAJA_ENABLE_CUDA)
+              std::is_same<REDUCE_POLICY,RAJA::cuda_reduce>::value
+              #elif defined(RAJA_ENABLE_HIP)
+              std::is_same<REDUCE_POLICY,RAJA::hip_reduce>::value
+              #elif defined(RAJA_ENABLE_SYCL)
+              std::is_same<REDUCE_POLICY,RAJA::sycl_reduce>::value
+              #else
+              false
+              #endif
+            )
```

Comment:
Can you factor out this logic into a constexpr function template or a class template? Then it won't have to be repeated twice.

Reply:
@MrBurmark Good call, I've turned it into a constexpr template function. What do you think?